lychee-lib/src/client.rs

```diff
@@ -141,13 +142,15 @@ impl ClientBuilder {
         })
         .build()?;
 
-        let github_token = match self.github_token {
-            Some(ref token) if !token.is_empty() => Some(Github::new(
-                self.user_agent.clone(),
-                Credentials::Token(token.clone()),
-            )?),
-            _ => None,
-        };
+        let github_token = self
+            .github_token
+            .as_ref()
+            .and_then(|token| {
```

Comment:
Again.
When I refactored these code, I deliberately avoided this kind of pattern.

Reply:
I agree that the `not()` should be a `!` but think this is way more readable than the `Some(ref token) if !token.is_empty() => Some(` incantation it replaced. I agree that this is a matter of opinion though.

Reply:
No.

I deliberately keep `Some(ref token)`. this make it very clear of the type signature.
( Calling `as_ref()` has the same effect. )

The not readable part is the nested `and_then` and `then`. Unless you add explicit type signature, it's less clear. But if you did, it's too verbose.